### ISSUE TITLE
drag n drop functionality

### ISSUE THREAD

**Issue by ShadOoW** [awesomeWM/awesome#604]
After searching and some testing, I wasn't able to find a way to handle drag n drop in a easy way

Present: Right now i have to drag then mod4+k/j then drop
Suggested: Drag then mouse enter client on tasklist then raise client then drop

Problem with second scenario is that mouse enter on tasklist doesn't fire when mouse button is pressed (drag)

How do you do drag n drop?

**Comment by osleg** [comment]
We don't....

**Comment by blueyed** [comment]
I agree that it would be nice if clients could be raised while drag'n'dropping, and hovering over a tasklist entry, or another client.
What I usually do in this case is to move the source client to some corner (floating), so that the target area is visible, and then drag'n'drop from there.
I have shortcuts to move a client to the top/bottom right corner(s) with different sizes.

**Comment by ShadOoW** [comment]
Sorry closed by mistake.

PS: I still have no solution for this

**Comment by HactarCE** [comment]
How about a `mouse::enter` callback on each tasklist item that activates its window if the mouse is held? Maybe only activate after the mouse hovers over it for a second or so?

The way to fix this properly would be to have Awesome implement Lua bindings for [XDND](https://freedesktop.org/wiki/Specifications/XDND/) (namely `XdndSelection`), and then a `mouse::enter` callback could check to see if there's a drag-and-drop happening.

**Comment by psychon** [comment]
> The way to fix this properly would be to have Awesome implement XDND (if it hasn't already) 

Yup, it has not.

**Comment by actionless** [comment]
> How about a mouse::enter callback on each tasklist item that activates its window if the mouse is held? Maybe only activate after the mouse hovers over it for a second or so?

and next to determine active client by https://awesomewm.org/apidoc/core_components/mouse.html#object_under_pointer

i didn't tried, but i think it should work
